flake.nix

```diff
@@ -3,6 +3,7 @@
 
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-22.05";
```

Comment:
Yeah, you're right.  When you pin to a release branch, you want to pin to `nixos-22.05`, not `release-22.05`.

`release-22.05` is the branch that stuff is merged into in Nixpkgs.  So if someone wants to backport something, or fix something in 22.05, they have to target the `release-22.05` branch.

And then the `nixos-22.05` branch is the most recent commit on `release-22.05` that has passed all tests on Hydra.  So `nixos-22.05` will always be a little older than `release-22.05`.

The danger with using `release-22.05` is that Hydra hasn't yet built some package you are using, so you end up having to build a bunch of things yourself.

If you look on https://status.nixos.org/, you can see the `nixos-22.05` channel is listed, which corresponds to the `nixos-22.05` branch in Nixpkgs.